extension.js

```diff
@@ -33,11 +33,26 @@ function format (document) {
       vscode.window.showInformationMessage('Do you have Purty installed? "npm install purty"');
     })
 }
-function purty (document) {
-  const localPurty = findLocalPurty(document.fileName);
-  const cmd = `${localPurty || 'purty'} -`;
+
+function purty(document) {
+  const configs = vscode.workspace.getConfiguration('purty');
+  let purtyCmd;
+  // We use empty string to mean unspecified because it means that the setting
+  // can be edited without having to write json (`["string", "null"]` does not
+  // have this property).
+  if (configs.pathToPurty !== "") {
+    purtyCmd = configs.pathToPurty;
+  } else {
+    const localPurty = findLocalPurty(document.fileName);
+    if (localPurty != null) {
+      purtyCmd = localPurty;
+    } else {
+      purtyCmd = 'purty';
+    }
+  }
```

Comment:
You could make a pure function `getPurtyCmd(configs, filename)` which would return the "purtyCmd" instead of mutating `purtyCmd` variable. It could be `const purtyCmd = getPurtyCmd(configs, document.filename)`. What do you say?